### ISSUE TITLE
Rename NonHelper to Intrinsic?

### ISSUE THREAD

**Issue by 0dvictor** [eclipse/omr#3050]
In the discussion of #2997 and #3049, as well as development of atomic operation support, I found the name of "NonHelper" is quite confusing. They look alike to helper methods but not a real helper because CodeGen is expected to generate inlined sequences instead of a helper call. Similar concept is known as "intrinsic" or "intrinsic method/function" in many other compilers.
Should we change the concept of "NonHelper" to "Intrinsic", or some other name that less confusing?

**Comment by 0dvictor** [comment]
For example change the enum `CommonNonhelperSymbol` to `CommonIntrinsicSymbol`.

**Comment by 0dvictor** [comment]
@0xdaryl @andrewcraik @fjeremic @Leonardo2718 @mstoodle @ymanton

**Comment by andrewcraik** [comment]
I think a question such as this deserves to go on the agenda of one of the architecture meetings. @0xdaryl

**Comment by fjeremic** [comment]
I agree the term "NonHelper" is confusing. I'm in favor of renaming to "Intrinsic". I also agree we should bring this up on the architecture meeting discussing the topic.

**Comment by 0xdaryl** [comment]
I'd like to move this discussion to today's architecture meeting: @0dvictor @ymanton @fjeremic @andrewcraik @gita-omr @Leonardo2718 @vijaysun-omr
